ctapipe/io/hessiofilereader.py

```diff
@@ -0,0 +1,211 @@
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+from ctapipe.core import Provenance
+from ctapipe.io.eventfilereader import EventFileReader
+from ctapipe.io.containers import DataContainer
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription
+
+
+class HessioFileReader(EventFileReader):
+    """
+    EventFileReader for the hessio data format.
+
+    This class utilises `pyhessio` to read the hessio file, and stores the
+    information into the event containers.
+    """
+    _count = 0
+
+    def __init__(self, config, tool, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        try:
+            from pyhessio import open_hessio
+            from pyhessio import close_file
+            from pyhessio import HessioGeneralError
+        except ImportError:
+            msg = "The `pyhessio` python module is required to access MC data"
+            self.log.error(msg)
+            raise
+
+        self.open_hessio = open_hessio
+        self.close_hessio = close_file
+        self.HessioGeneralError = HessioGeneralError
+
+        self.allowed_tels = None
+
+        if HessioFileReader._count > 0:
+            self.log.warn("Only one pyhessio reader allowed at a time. "
+                          "Previous hessio file will be closed.")
+            self.close_hessio()
+        HessioFileReader._count += 1
+
+        self._metadata['is_simulation'] = True
+
+    @staticmethod
+    def is_compatible(file_path):
+        return file_path.endswith('.gz')
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close_hessio()
+
+    def _generator(self):
+        with self.open_hessio(self.input_path) as pyhessio:
+            # the container is initialized once, and data is replaced within
+            # it after each yield
+            Provenance().add_input_file(self.input_path, role='dl0.sub.evt')
+            counter = 0
+            eventstream = pyhessio.move_to_next_event()
+            if self.allowed_tels is not None:
+                self.allowed_tels = set(self.allowed_tels)
+            data = DataContainer()
+            data.meta['origin'] = "hessio"
```

Comment:
What are people supposed to do with `data.meta["origin"]`? Could you give an example somewhere?

Reply:
It could be removed as the reading is explicitly attached to an EventFileReader now (which explicitly tells you the "origin")